storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java

```diff
@@ -348,14 +350,27 @@ public void doExecutorHeartbeats() {
         if (null == executors) {
             stats = ClientStatsUtil.mkEmptyExecutorZkHbs(workerState.localExecutors);
         } else {
-            stats = ClientStatsUtil.convertExecutorZkHbs(executors.stream().collect(Collectors
-                                                                                  .toMap(IRunningExecutor::getExecutorId,
-                                                                                         IRunningExecutor::renderStats)));
+            stats = ClientStatsUtil.convertExecutorZkHbs(
+                    executors.stream().collect(
+                            Collectors.toMap(
+                                    IRunningExecutor::getExecutorId,
+                                    IRunningExecutor::renderStats
+                            )
+                    )
+            );
         }
-        Map<String, Object> zkHB = ClientStatsUtil.mkZkWorkerHb(workerState.topologyId, stats, workerState.uptime.upTime());
+
+        Map<String, Object> zkHB = ClientStatsUtil.mkZkWorkerHb(
+                workerState.topologyId, stats, workerState.uptime.upTime()
+        );
+
         try {
+            String assignmentId = workerState.assignmentId;
+            if (this.numaId != null) {
+                assignmentId += Constants.NUMA_ID_SEPARATOR + this.numaId;
```

Comment:
I've tested this out - the worker will launch correctly but wont' report back to nimbus and from the UI it will appear the workers haven't started yet.